### DISCUSSION TITLE
Problem with on-demand ISR

### DISCUSSION THREAD

**Discussion by pedro-lucca** [vercel/next.js#34773] [Help]
Hi Guys, i`m trying implement on-demand ISR in my code, but doesn't work at all!!

I trying revalidate a path per call, but aways return (Error revalidating)!

Obs: Localhost, using npm run build and start, work fine.

`
export default async function handler(req, res) {
  if (req.query.secret !== process.env.MY_SECRET_TOKEN) {
    return res.status(401).json({ message: 'Invalid token' })
  }
  if (!req.query.slug) {
    return res.status(400).json({ message: 'slug is required' })
  }
  try {
    const slug = req.query.slug
    await res.unstable_revalidate('/', '/article/' + slug)
    return res.json(
      {
        revalidated: true,
        slug: slug || null,
      })
  } catch (err) {
    return res.status(500).send('Error revalidating')
  }
}
`

**Comment by ijjk** [comment]
Hi, can you provide additional info a reproduction/deployment where you are seeing this occur?

**Reply by pedro-lucca** [reply]
i`m using vercel, but i try now using this code and work fine

export default async function handler(req, res) {

  const slug = req.query.slug;
  
  try {
    await res.unstable_revalidate('/');
    if (slug) {
      await res.unstable_revalidate(slug);
    }
    return res.json({ revalidated: true })
  } catch (err) {
    return res.status(500).send('Error revalidating')
  }
}

**Reply by pedro-lucca** [reply]
i think the problem is use await res.unstable_revalidate(`article/${slug}`);

**Reply by pedro-lucca** [reply]
hi, i found the error, correction, my error LOL

correct is use await res.unstable_revalidate(`/article/${slug}`);

i forgot / before article path

**Reply by ijjk** [reply]
Ah glad you narrowed in on the issue
